### PR TITLE
Svyatoslav/feature/pollock bot

### PR DESCRIPTION
**Pull request by SvyatoslavArtymyvych**

### CODE REVIEW

File: app/controllers/hander_chat_member.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
+from sqlalchemy import and_
+from aiogram.types import ContentType
+
+from .pollock_bot_listener import BotListener
+from app.models import User, Chat, ChatUser
+from app.logger import log
```
Comment (varvarashcherbyna7) [inline app/controllers/hander_chat_member.py:6]: modules need to be imported after aiogram.types

File: app/controllers/handler_photo_contest.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
+from datetime import datetime
+
+from .pollock_bot_listener import BotListener
+from app.models import PhotoContest, Photo
+from app.logger import log
```
Comment (varvarashcherbyna7) [inline app/controllers/handler_photo_contest.py:5]: from app.logger import log 
import after datetime

File: app/controllers/handler_photo_contest.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    async def fire_message(self, message):
+        await self.add_photo_to_contest(message)
+
+    def validate_data(self, contest_tag):
+        contest_tag = contest_tag
+        pc = PhotoContest.query.filter(PhotoContest.hashtag == contest_tag).first()
+        date_now = datetime.utcnow()
+
+        if pc:
+            start_date = pc.start_date
```
Comment (varvarashcherbyna7) [inline app/controllers/handler_photo_contest.py:18]: after if pc
add log INFO that PhotoContest exists

File: app/controllers/handler_text.py [hunk 4]
```diff
@@ -0,0 +1,27 @@
+from aiogram import types
+
+from .pollock_bot_listener import BotListener
+from app.logger import log
```
Comment (varvarashcherbyna7) [inline app/controllers/handler_text.py:4]: import log after aiogram

File: app/controllers/pollock_bot.py [hunk 5]
```diff
@@ -1,213 +1,84 @@
...
-                    log.INFO,
-                    "BOT: on_left_chat_member: link on chat_id [%d] for user_id [%d] deleted ",
-                    chat_user.chat_id,
-                    chat_user.user_id,
-                )
+            if message.content_type in self.listeners:
+                for listener in self.listeners[message.content_type]:
+                    await listener.fire_message(message)
 
     def run(self):
```
Comment (varvarashcherbyna7) [inline app/controllers/pollock_bot.py:47]: this function must be at the end

File: app/controllers/pollock_bot_listener.py [hunk 6]
```diff
@@ -0,0 +1,96 @@
+
+from sqlalchemy import and_
+
+from app.models import User, Chat, ChatUser
+from app.logger import log
```
Comment (varvarashcherbyna7) [inline app/controllers/pollock_bot_listener.py:5]: log import after sqlalchemy

File: app/controllers/pollock_bot_listener.py [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+    def user_from_db(self, message):
+        log(log.INFO, "BOT: Get user from db")
+
+        t_user = message.from_user
+        user = User.query.filter(User.telegram_id == t_user.id).first()
+        if not user:
+            user = register(id=t_user.id)
+        return user
+
+    async def is_user_is_member(self, user_id, chat_id):
```
Comment (varvarashcherbyna7) [inline app/controllers/pollock_bot_listener.py:63]: replace with is_user_member

**Review by varvarashcherbyna7 — CHANGES_REQUESTED** [review]

**Review by kipod — APPROVED** [review]

**Pull request merged**